src/components/Toolbar/ToolbarFind.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { Button, Icon } from '../../index';
+import { bindMethods } from '../../common/helpers';
+
+export class ToolbarFind extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      dropdownShown: false,
+      currentValue: ''
+    };
+
+    bindMethods(this, [
+      'toggleDropdownShown',
+      'hideDropdown',
+      'onValueKeyPress',
+      'handleValueChange',
+      'handleFindNext',
+      'handleFindPrevious'
+    ]);
+  }
+
+  toggleDropdownShown() {
+    const { dropdownShown } = this.state;
+    this.setState({ dropdownShown: !dropdownShown });
+  }
+
+  hideDropdown() {
+    this.setState({ dropdownShown: false });
+  }
+
+  onValueKeyPress(keyEvent) {
+    const { onEnter } = this.props;
+    const { currentValue } = this.state;
+
+    if (keyEvent.key === 'Enter' && onEnter) {
+      onEnter(currentValue);
+    }
+  }
+
+  handleValueChange(event) {
+    const { onChange } = this.props;
+
+    this.setState({ currentValue: event.target.value });
+
+    if (onChange) {
+      onChange(event.target.value);
+    }
+  }
+
+  handleFindNext() {
+    const { currentValue } = this.state;
+    const { onFindNext } = this.props;
+
+    if (onFindNext) {
+      onFindNext(currentValue);
+    }
+  }
+
+  handleFindPrevious() {
+    const { currentValue } = this.state;
+    const { onFindPrevious } = this.props;
+
+    if (onFindPrevious) {
+      onFindPrevious(currentValue);
+    }
+  }
+
+  renderCounts() {
+    const { currentValue } = this.state;
+    const { currentIndex, totalCount } = this.props;
+
+    if (currentValue && currentValue !== '') {
+      return [
+        <span className="find-pf-nums">
```

Comment:
maybe the span and buttons can be combined into another stateless component <Counts> ? Not sure how common it is...

Reply:
I'd prefer to keep it all self contained unless you feel strongly.

Reply:
fine w/ me, just wasn't sure how common. ;)